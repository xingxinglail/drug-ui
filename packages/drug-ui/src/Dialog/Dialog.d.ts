import * as React from 'react';
import { SimpleSpread } from '..';
interface PropsExtra {
    title?: React.ReactNode | string;
}
export interface IProps extends SimpleSpread<React.HTMLAttributes<HTMLDivElement>, PropsExtra> {
    visible?: boolean;
    zIndex?: number;
    width?: string | number;
    top?: string | number;
    footer?: React.ReactNode | string;
    mask?: boolean;
    maskClosable?: boolean;
    closable?: boolean;
    keepMounted?: boolean;
    transitionDuration?: number | {
        enter?: number;
        exit?: number;
    };
    ref?: React.Ref<HTMLDivElement>;
    onClose?: (e: React.MouseEvent) => void;
    onEnter?: (e: HTMLElement) => void;
    onEntering?: (e: HTMLElement) => void;
    onEntered?: (e: HTMLElement) => void;
    onExit?: (e: HTMLElement) => void;
    onExiting?: (e: HTMLElement) => void;
    onExited?: (e: HTMLElement) => void;
}
declare const Dialog: React.FC<IProps>;
export default Dialog;
