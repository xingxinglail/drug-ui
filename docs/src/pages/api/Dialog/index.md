---
filename: /packages/drug-ui/src/Dialog/Dialog.js
---

# Dialog API

<p class="description">Dialog React 组件的 API 文档。</p>

## Import

```js
import Dialog from '@drug-ui/core/Dialog';
// or
import { Dialog } from '@drug-ui/core';
```

## Props

| Name | Type | Default | Description |
|:-----|:-----|:--------|:------------|
| <span class="prop-name">children</span> | <span class="prop-type">node</span> |  | 组件内容。 |
| <span class="prop-name">visible</span> | <span class="prop-type">bool</span> | <span class="prop-default">true</span> | 如果为 ```true```，则显示组件。 |
| <span class="prop-name">zIndex</span> | <span class="prop-type">number</span> | <span class="prop-default">1000</span> | ```z-index``` 层级样式。 |
| <span class="prop-name">width</span> | <span class="prop-type">string &#124; number</span> | <span class="prop-default">50%</span> | 组件宽度。 |
| <span class="prop-name">top</span> | <span class="prop-type">string &#124; number</span> | <span class="prop-default">15vh</span> | 组件距顶部位置。 |
| <span class="prop-name">mask</span> | <span class="prop-type">bool</span> | <span class="prop-default">true</span> | 如果为 ```true```，则显示遮罩。 |
| <span class="prop-name">maskClosable</span> | <span class="prop-type">bool</span> | <span class="prop-default">true</span> | 如果为 ```true```，则可以点击遮罩关闭对话框。 |
| <span class="prop-name">closable</span> | <span class="prop-type">bool</span> | <span class="prop-default">true</span> | 如果为 ```true```，则显示关闭图标。 |
| <span class="prop-name">keepMounted</span> | <span class="prop-type">bool</span> | <span class="prop-default">true</span> | 如果为 ```true```，则不会被销毁。 |
| <span class="prop-name">footer</span> | <span class="prop-type">string &#124; node</span> | <span class="prop-default"></span> | 对话框底部内容。 |
| <span class="prop-name">transitionDuration</span> | <span class="prop-type">number &#124; { enter?: number, exit?: number }</span> | <span class="prop-default">300</span> | 过渡的持续时间（以毫秒为单位）。 |
| <span class="prop-name">onClose</span> | <span class="prop-type">func</span> | <span class="prop-default"></span> | 当组件请求关闭时触发回调。 |
| <span class="prop-name">onEnter</span> | <span class="prop-type">func</span> | <span class="prop-default"></span> | 对话框进入之前触发了回调。 |
| <span class="prop-name">onEntering</span> | <span class="prop-type">func</span> | <span class="prop-default"></span> | 当对话框进入时触发回调。 |
| <span class="prop-name">onEntered</span> | <span class="prop-type">func</span> | <span class="prop-default"></span> | 进入对话框后触发回调。 |
| <span class="prop-name">onExit</span> | <span class="prop-type">func</span> | <span class="prop-default"></span> | 对话框退出前触发了回调。 |
| <span class="prop-name">onExiting</span> | <span class="prop-type">func</span> | <span class="prop-default"></span> | 对话框退出时触发回调。 |
| <span class="prop-name">onExited</span> | <span class="prop-type">func</span> | <span class="prop-default"></span> | 对话框退出后触发回调。 |

`ref` 被转发到根元素.

## CSS

Style sheet name: `DuiDialog`.

## Demos

- [Dialog](/drug-ui/components/Dialogs)




