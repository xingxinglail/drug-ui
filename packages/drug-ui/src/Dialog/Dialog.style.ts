import { Style, Styles } from 'jss';
import { Theme } from '../styles';

const fixed = {
    position: 'fixed',
    top: 0,
    right: 0,
    bottom: 0,
    left: 0
};

const root: Style = fixed;

const container: Style = {
    ...fixed,
    overflow: 'auto'
};

const paper: Style = {
    position: 'relative',
    margin: '0 auto 50px',
    borderRadius: 4,
    backgroundColor: '#fff',
    boxShadow: '0 11px 15px -7px rgba(0, 0, 0, 0.2), 0 24px 38px 3px rgba(0, 0, 0, 0.14), 0 9px 46px 8px rgba(0, 0, 0, 0.12)'
};

const header: Style = {
    flex: '0 0 auto',
    margin: 0,
    padding: [16, 24],
    '& h2': {
        fontSize: 20,
        fontWeight: 500,
        lineHeight: 1.6,
        letterSpacing: '0.0075em'
    }
};

const close: Style = {
    position: 'absolute',
    top: 12,
    right: 12,
    '& svg': {
        fontSize: 14,
        color: '#9e9e9e'
    }
};

const body: Style = {
    padding: [16, 24],
    borderTop: '1px solid rgba(0, 0, 0, 0.12)',
    borderBottom: '1px solid rgba(0, 0, 0, 0.12)'
};

const footer: Style = {
    display: 'flex',
    padding: 8,
    alignItems: 'center',
    justifyContent: 'flex-end'
};

const mask: Style = {
    ...fixed,
    backgroundColor: 'rgba(0, 0, 0, 0.5)',
    zIndex: -1
};

export const styles = (_: Theme): Styles => ({
    root,
    container,
    paper,
    header,
    body,
    close,
    footer,
    mask
});
