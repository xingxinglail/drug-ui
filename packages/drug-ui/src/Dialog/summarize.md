### Dialog 总结

todo
1. Exclude
    -

2. Pick
    -

3. SimpleSpread
    -

4. useCombinedRefs
    -

5. react-transition-group/Transition
    -

