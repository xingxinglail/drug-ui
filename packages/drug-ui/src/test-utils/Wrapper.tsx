import * as React from 'react';
import { ThemeProvider } from '../index';

interface Props {
    children?: any;
}

const Wrapper: React.FC<Props> = (props) => {
    return (
        <ThemeProvider>
            { props.children }
        </ThemeProvider>
    );
};

export default Wrapper;
