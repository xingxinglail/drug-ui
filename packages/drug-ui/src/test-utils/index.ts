export { default as Wrapper } from './Wrapper';
export { getClasses } from './getClasses';

export const wait = (delay = 50) => {
    return new Promise(resolve => setTimeout(resolve, delay));
};
