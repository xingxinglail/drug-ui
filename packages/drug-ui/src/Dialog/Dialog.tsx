import * as React from 'react';
import * as ReactDOM from 'react-dom';
import classnames from 'classnames';
import * as PropTypes from 'prop-types';
import { Transition } from 'react-transition-group';
import { IconButton, SimpleSpread } from '@drug-ui/core';
import { createUseStyles } from '@drug-ui/core/styles';
import { useCombinedRefs } from '@drug-ui/hooks';
import { Close } from '@drug-ui/icons';
import { styles } from './Dialog.style';

interface PropsExtra {
    title?: React.ReactNode | string;
}

export interface IProps extends SimpleSpread<React.HTMLAttributes<HTMLDivElement>, PropsExtra> {
    visible?: boolean;
    zIndex?: number;
    width?: string | number;
    top?: string | number;
    footer?: React.ReactNode | string;
    mask?: boolean;
    maskClosable?: boolean;
    closable?: boolean;
    keepMounted?: boolean;
    transitionDuration?: number | { enter?: number, exit?: number };
    ref?: React.Ref<HTMLDivElement>;
    onClose?: (e: React.MouseEvent) => void;
    onEnter?: (e: HTMLElement) => void;
    onEntering?: (e: HTMLElement) => void;
    onEntered?: (e: HTMLElement) => void;
    onExit?: (e: HTMLElement) => void;
    onExiting?: (e: HTMLElement) => void;
    onExited?: (e: HTMLElement) => void;
}

type LayoutClassProps = 'root' | 'mask' | 'container' | 'paper' | 'header' | 'close' | 'body' | 'footer';

const name = 'Dialog';

const useStyles = createUseStyles<LayoutClassProps>(styles, name);

const maskTransitionStyles = {
    entering: { opacity: 1 },
    entered: { opacity: 1 },
    exiting: { opacity: 0 },
    exited: { opacity: 0 },
};

const maskDefaultStyle = {
    transition: 'opacity 300ms',
    opacity: 0
};

const paperDefaultStyle = {
    transition: 'transform 300ms, opacity 300ms',
    opacity: 0,
    transform: 'translate3d(0, -20px, 0)',
};

const paperTransitionStyles = {
    entering: { opacity: 1, transform: 'translate3d(0, 0, 0)' },
    entered: { opacity: 1, transform: 'translate3d(0, 0, 0)' },
    exiting: { opacity: 0, transform: 'translate3d(0, -20px, 0)' },
    exited: { opacity: 0, transform: 'translate3d(0, -20px, 0)' },
};

const Portal: React.FC = props => {
    return ReactDOM.createPortal(
        props.children,
        document.body
    );
};

const Dialog: React.FC<IProps> = React.forwardRef<HTMLDivElement, IProps>((props, ref) => {
    const { visible, className, keepMounted, title, footer, width, zIndex, top, children, mask, maskClosable, closable, onClose, transitionDuration = 300, ...rest } = props;
    const [exited, setExited] = React.useState(true);
    const classes = useStyles();
    const innerRef = React.useRef<HTMLDivElement>(null);
    const combinedRef = useCombinedRefs<HTMLDivElement>(ref, innerRef);
    const classNames = classnames(
        classes.root,
        className
    );

    const closeHandle = (e: React.MouseEvent) => {
        onClose && onClose(e);
    };

    const paperHandleClick = (e: React.MouseEvent) => {
        e.stopPropagation();
    };

    React.useLayoutEffect(() => {
        // So the animation always start from the start.
        innerRef.current?.scrollTop;
        document.body.style.overflow = visible ? 'hidden' : '';
    }, [visible]);

    if (!keepMounted && !visible && exited) return null;

    const handleExitTransitionDuration = (node: HTMLElement, type: string) => {
        const paper: HTMLDivElement | null = node.querySelector(`.${ classes.paper }`);
        const mask: HTMLDivElement | null = node.querySelector(`.${ classes.mask }`);
        const duration = `${ typeof transitionDuration === 'number' ? transitionDuration : transitionDuration[type] }ms`;
        if (mask) mask.style.transitionDuration = duration;
        if (paper) paper.style.transitionDuration = duration;
    };

    const handleTransitionEvent = (event: string, e: HTMLElement) => {
        if (event === 'onEnter') {
            setExited(false);
            handleExitTransitionDuration(e, 'enter');
        }
        if (event === 'onExit') handleExitTransitionDuration(e, 'exit');
        if (event === 'onExited') setExited(true);
        rest[event] && rest[event](e);
    };

    return (
        <Portal>
            <div
                className={ classNames }
                style={ { zIndex, visibility: (!visible && exited) ? 'hidden' : 'visible' } }
                ref={ combinedRef }>
                <Transition
                    appear
                    in={ visible }
                    timeout={ transitionDuration }
                    onEnter={ e => handleTransitionEvent('onEnter', e) }
                    onEntering={ e => handleTransitionEvent('onEntering', e) }
                    onEntered={ e => handleTransitionEvent('onEntered', e) }
                    onExit={ e => handleTransitionEvent('onExit', e) }
                    onExiting={ e => handleTransitionEvent('onExiting', e) }
                    onExited={ e => handleTransitionEvent('onExited', e) }>
                    { state => (
                        <div>
                            { !mask ? null :
                                <div
                                    className={ classes.mask }
                                    style={ { ...maskDefaultStyle, ...maskTransitionStyles[state] } } />
                            }
                            <div
                                className={ classes.container }
                                onClick={ e => maskClosable && closeHandle(e) }>
                                <div
                                    className={ classes.paper }
                                    style={ {
                                        width,
                                        marginTop: top, ...paperDefaultStyle, ...paperTransitionStyles[state]
                                    } }
                                    onClick={ paperHandleClick }>
                                    {
                                        !closable ? null :
                                            <div
                                                className={ classes.close }
                                                onClick={ closeHandle }>
                                                <IconButton size="small">
                                                    <Close />
                                                </IconButton>
                                            </div>
                                    }
                                    {
                                        !title ? null :
                                            <div className={ classes.header }>
                                                <h2>{ title }</h2>
                                            </div>
                                    }
                                    <div className={ classes.body }>{ children }</div>
                                    { footer ? <div className={ classes.footer }>{ footer }</div> : null }
                                </div>
                            </div>
                        </div>
                    ) }
                </Transition>
            </div>
        </Portal>
    );
});

Dialog.displayName = name;
Dialog.defaultProps = {
    visible: false,
    title: '',
    zIndex: 1000,
    width: '50%',
    top: '15vh',
    mask: true,
    maskClosable: true,
    closable: true,
    keepMounted: false,
    transitionDuration: 300,
    onClose: undefined,
    onEnter: undefined,
    onEntering: undefined,
    onEntered: undefined,
    onExit: undefined,
    onExiting: undefined,
    onExited: undefined
};

Dialog.propTypes = {
    className: PropTypes.string,
    visible: PropTypes.bool,
    title: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.element
    ]),
    zIndex: PropTypes.number,
    width: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number
    ]),
    top: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number
    ]),
    mask: PropTypes.bool,
    maskClosable: PropTypes.bool,
    closable: PropTypes.bool,
    keepMounted: PropTypes.bool,
    transitionDuration: PropTypes.oneOfType([
        PropTypes.number,
        PropTypes.object
    ]),
    onClose: PropTypes.func,
    onEnter: PropTypes.func,
    onEntering: PropTypes.func,
    onEntered: PropTypes.func,
    onExit: PropTypes.func,
    onExiting: PropTypes.func,
    onExited: PropTypes.func
};

export default Dialog;

