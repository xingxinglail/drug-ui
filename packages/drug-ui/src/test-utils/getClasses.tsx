import * as React from 'react';
import * as enzyme from 'enzyme';
import Wrapper from './Wrapper';

export const getClasses = (element: React.ReactElement) => {
    // @ts-ignore
    const { useStyles } = element.type;
    let classes;

    function Listener () {
        classes = useStyles();
        return null;
    }

    enzyme.mount(<Wrapper><Listener /></Wrapper>);
    return classes;
};
