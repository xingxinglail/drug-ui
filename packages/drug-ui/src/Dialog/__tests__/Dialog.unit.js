import * as React from 'react';
import * as enzyme from 'enzyme';
import EnzymeToJson from 'enzyme-to-json';
import Dialog from '..';
import { Wrapper } from '@drug-ui/core/test-utils';

const mount = enzyme.mount;

describe('<Dialog />', () => {

    it('render successful', () => {
        const subject = mount(<Wrapper><Dialog visible={ true } title="title">Dialog</Dialog></Wrapper>);
        expect(EnzymeToJson(subject)).toMatchSnapshot();
    });

    it('设置 className', () => {
        const wrapper = mount(<Wrapper><Dialog visible={ true } className="my-dialog">Dialog</Dialog></Wrapper>);
        const classList = wrapper.getDOMNode().classList;
        expect(classList.contains('DuiDialog-root')).toBeTruthy();
        expect(classList.contains('my-dialog')).toBeTruthy();
    });

    it('设置 zIndex', () => {
        const wrapper = mount(<Wrapper><Dialog visible={ true } zIndex={ 1111 }>Dialog</Dialog></Wrapper>);
        expect(getComputedStyle(wrapper.getDOMNode()).zIndex).toBe('1111');
    });

    it('设置 width', () => {
        const wrapper = mount(<Wrapper><Dialog visible={ true } width={ 400 }>Dialog</Dialog></Wrapper>);
        const dom = wrapper.getDOMNode();
        const paper = dom.querySelector('.DuiDialog-paper');
        expect(getComputedStyle(paper).width).toBe('400px');
        wrapper.setProps({ width: '33%' });
        expect(getComputedStyle(paper).width).toBe('33%');
    });

    it('设置 top', () => {
        const wrapper = mount(<Wrapper><Dialog visible={ true } top={ 28 }>Dialog</Dialog></Wrapper>);
        const dom = wrapper.getDOMNode();
        const paper = dom.querySelector('.DuiDialog-paper');
        expect(getComputedStyle(paper).marginTop).toBe('28px');
        wrapper.setProps({ top: '55%' });
        expect(getComputedStyle(paper).marginTop).toBe('55%');
    });

    it('设置 title', () => {
        const wrapper = mount(<Wrapper><Dialog visible={ true }>Dialog</Dialog></Wrapper>);
        const dom = wrapper.getDOMNode();
        expect(wrapper.find('.DuiDialog-header').exists()).toBeFalsy();
        wrapper.setProps({ title: 'Title' });
        expect(wrapper.find('.DuiDialog-header').exists()).toBeTruthy();
        const Title = <div className="node-title">Title</div>;
        wrapper.setProps({ title: Title });
        expect(dom.querySelector('.node-title')).toBeTruthy();
    });

    it('设置 footer', () => {
        const wrapper = mount(<Wrapper><Dialog visible={ true }>Dialog</Dialog></Wrapper>);
        const dom = wrapper.getDOMNode();
        expect(wrapper.find('.DuiDialog-footer').exists()).toBeFalsy();
        wrapper.setProps({ footer: 'Footer' });
        expect(wrapper.find('.DuiDialog-footer').exists()).toBeTruthy();
        const Title = <div className="node-footer">Title</div>;
        wrapper.setProps({ title: Title });
        expect(dom.querySelector('.node-footer')).toBeTruthy();
    });

    it('设置 mask', () => {
        const wrapper = mount(<Wrapper><Dialog visible={ true }>Dialog</Dialog></Wrapper>);
        const dom = wrapper.getDOMNode();
        expect(dom.querySelector('.DuiDialog-mask')).toBeTruthy();
        wrapper.setProps({ mask: false });
        expect(dom.querySelector('.DuiDialog-mask')).toBeFalsy();
    });

    it('设置 maskClosable', () => {
        const fn = jest.fn();
        const wrapper = mount(<Wrapper><Dialog visible={ true } onClose={ fn }>Dialog</Dialog></Wrapper>);
        const dom = wrapper.getDOMNode();
        const container = dom.querySelector('.DuiDialog-container');
        container.click();
        expect(fn).toBeCalled();
        wrapper.setProps({ maskClosable: false });
        container.click();
        expect(fn).toBeCalledTimes(1);
    });

    it('设置 closable', () => {
        const fn = jest.fn();
        const wrapper = mount(<Wrapper><Dialog visible={ true } onClose={ fn }>Dialog</Dialog></Wrapper>);
        const dom = wrapper.getDOMNode();
        const closeIcon = dom.querySelector('.DuiDialog-close');
        closeIcon.click();
        expect(fn).toBeCalled();
        wrapper.setProps({ closable: false });
        closeIcon.click();
        expect(fn).toBeCalledTimes(1);
        expect(dom.querySelector('.DuiDialog-close')).toBeFalsy();
    });

    it('设置 keepMounted', done => {
        const onExited = () => {
            setTimeout(() => {
                expect(wrapper.getDOMNode()).toBeTruthy();
                done();
            });
        };
        const wrapper = mount(<Wrapper><Dialog onExited={ onExited }>Dialog</Dialog></Wrapper>);
        wrapper.setProps({ visible: true });
        expect(wrapper.getDOMNode()).toBeTruthy();
        wrapper.setProps({ keepMounted: true });
        wrapper.setProps({ visible: false });
    });

    it('设置 transitionDuration', done => {
        let wrapper = null;
        let n = 1;
        let n2 = 1;
        const onEnter = node => {
            const paper = node.querySelector('.DuiDialog-mask');
            const mask = node.querySelector('.DuiDialog-paper');
            if (n === 1) {
                expect(getComputedStyle(paper).transitionDuration).toBe('100ms');
                expect(getComputedStyle(mask).transitionDuration).toBe('100ms');
                n++
            } else {
                expect(getComputedStyle(paper).transitionDuration).toBe('111ms');
                expect(getComputedStyle(mask).transitionDuration).toBe('111ms');
            }
            wrapper.setProps({ visible: false });
        }

        const onExited = node => {
            const paper = node.querySelector('.DuiDialog-mask');
            const mask = node.querySelector('.DuiDialog-paper');
            if (n2 === 1) {
                expect(getComputedStyle(paper).transitionDuration).toBe('100ms');
                expect(getComputedStyle(mask).transitionDuration).toBe('100ms');
                n2++;
                wrapper.setProps({ transitionDuration: { enter: 111, exit: 222 } });
                wrapper.setProps({ visible: true });
            } else {
                expect(getComputedStyle(paper).transitionDuration).toBe('222ms');
                expect(getComputedStyle(mask).transitionDuration).toBe('222ms');
                done();
            }
        };
        wrapper = mount(
            <Wrapper>
                <Dialog
                    transitionDuration={ 100 }
                    onEnter={ onEnter }
                    onExited={ onExited }>
                    Dialog
                </Dialog>
            </Wrapper>
        );
        wrapper.setProps({ visible: true });
    });

    it('Transition 事件触发', done => {
        const fn = jest.fn();
        const wrapper = mount(
            <Wrapper>
                <Dialog
                    onEnter={ fn }
                    onEntering={ fn }
                    onEntered={ fn }
                    onExit={ fn }
                    onExiting={ fn }
                    onExited={ fn }>Dialog</Dialog>
            </Wrapper>);
        wrapper.setProps({ visible: true });
        setTimeout(() => {
            const dom = wrapper.getDOMNode().children[0];
            wrapper.setProps({ visible: false });
            setTimeout(() => {
                const calls = fn.mock.calls;
                expect(calls[0][0]).toBe(dom);
                expect(calls[1][0]).toBe(dom);
                expect(calls[2][0]).toBe(dom);
                expect(calls[3][0]).toBe(dom);
                expect(calls[4][0]).toBe(dom);
                expect(calls[5][0]).toBe(dom);
                expect(calls.length).toBe(6);
                done();
            }, 310);
        }, 310);
    });
});
