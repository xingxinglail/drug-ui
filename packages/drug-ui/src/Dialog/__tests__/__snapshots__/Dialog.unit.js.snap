// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dialog /> render successful 1`] = `
<Wrapper>
  <ThemeProvider>
    <JssProvider
      generateId={[Function]}
    >
      <ThemeProvider
        theme={
          Object {
            "palette": Object {
              "primary": Object {
                "contrastText": "#fff",
                "main": "#1976d2",
              },
              "secondary": Object {
                "contrastText": "#fff",
                "main": "#dc004e",
              },
            },
            "typography": Object {
              "fontSize": Object {
                "large": 15,
                "medium": 14,
                "small": 13,
              },
            },
          }
        }
      >
        <Dialog
          closable={true}
          keepMounted={false}
          mask={true}
          maskClosable={true}
          title="title"
          top="15vh"
          transitionDuration={300}
          visible={true}
          width="50%"
          zIndex={1000}
        >
          <Portal>
            <Portal
              containerInfo={
                <body
                  style="overflow: hidden;"
                >
                  <div
                    class="DuiDialog-root"
                    style="z-index: 1000; visibility: visible;"
                  >
                    <div>
                      <div
                        class="DuiDialog-mask"
                        style="transition: opacity 300ms; opacity: 1; transition-duration: 300ms;"
                      />
                      <div
                        class="DuiDialog-container"
                      >
                        <div
                          class="DuiDialog-paper"
                          style="width: 50%; margin-top: 15vh; transition: transform 300ms, opacity 300ms; opacity: 1; transform: translate3d(0, 0, 0); transition-duration: 300ms;"
                        >
                          <div
                            class="DuiDialog-close"
                          >
                            <button
                              class="DuiButtonBase-root DuiIconButton-root DuiIconButton-sizeSmall"
                            >
                              <span
                                class="DuiRipple-root"
                              />
                            </button>
                          </div>
                          <div
                            class="DuiDialog-header"
                          >
                            <h2>
                              title
                            </h2>
                          </div>
                          <div
                            class="DuiDialog-body"
                          >
                            Dialog
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="DuiDialog-root"
                    style="z-index: 1000; visibility: visible;"
                  >
                    <div>
                      <div
                        class="DuiDialog-mask"
                        style="transition: opacity 300ms; opacity: 1; transition-duration: 300ms;"
                      />
                      <div
                        class="DuiDialog-container"
                      >
                        <div
                          class="DuiDialog-paper"
                          style="width: 50%; margin-top: 15vh; transition: transform 300ms, opacity 300ms; opacity: 1; transform: translate3d(0, 0, 0); transition-duration: 300ms;"
                        >
                          <div
                            class="DuiDialog-close"
                          >
                            <button
                              class="DuiButtonBase-root DuiIconButton-root DuiIconButton-sizeSmall"
                            >
                              <span
                                class="DuiRipple-root"
                              />
                            </button>
                          </div>
                          <div
                            class="DuiDialog-header"
                          >
                            <h2>
                              title
                            </h2>
                          </div>
                          <div
                            class="DuiDialog-body"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </body>
              }
            >
              <div
                className="DuiDialog-root"
                style={
                  Object {
                    "visibility": "visible",
                    "zIndex": 1000,
                  }
                }
              >
                <Transition
                  appear={true}
                  enter={true}
                  exit={true}
                  in={true}
                  mountOnEnter={false}
                  onEnter={[Function]}
                  onEntered={[Function]}
                  onEntering={[Function]}
                  onExit={[Function]}
                  onExited={[Function]}
                  onExiting={[Function]}
                  timeout={300}
                  unmountOnExit={false}
                >
                  <div>
                    <div
                      className="DuiDialog-mask"
                      style={
                        Object {
                          "opacity": 1,
                          "transition": "opacity 300ms",
                        }
                      }
                    />
                    <div
                      className="DuiDialog-container"
                      onClick={[Function]}
                    >
                      <div
                        className="DuiDialog-paper"
                        onClick={[Function]}
                        style={
                          Object {
                            "marginTop": "15vh",
                            "opacity": 1,
                            "transform": "translate3d(0, 0, 0)",
                            "transition": "transform 300ms, opacity 300ms",
                            "width": "50%",
                          }
                        }
                      >
                        <div
                          className="DuiDialog-close"
                          onClick={[Function]}
                        >
                          <IconButton
                            centerRipple={true}
                            color="default"
                            size="small"
                          >
                            <ButtonBase
                              centerRipple={true}
                              className="DuiIconButton-root DuiIconButton-sizeSmall"
                              component="button"
                              disableRipple={false}
                              disabled={false}
                            >
                              <button
                                className="DuiButtonBase-root DuiIconButton-root DuiIconButton-sizeSmall"
                                disabled={false}
                              >
                                <Ripple
                                  center={true}
                                >
                                  <span
                                    className="DuiRipple-root"
                                    onDragLeave={[Function]}
                                    onMouseDown={[Function]}
                                    onMouseLeave={[Function]}
                                    onMouseUp={[Function]}
                                  />
                                </Ripple>
                              </button>
                            </ButtonBase>
                          </IconButton>
                        </div>
                        <div
                          className="DuiDialog-header"
                        >
                          <h2>
                            title
                          </h2>
                        </div>
                        <div
                          className="DuiDialog-body"
                        >
                          <Dialog
                            closable={true}
                            keepMounted={false}
                            mask={true}
                            maskClosable={true}
                            title="title"
                            top="15vh"
                            transitionDuration={300}
                            visible={true}
                            width="50%"
                            zIndex={1000}
                          >
                            <Portal>
                              <Portal
                                containerInfo={
                                  <body
                                    style="overflow: hidden;"
                                  >
                                    <div
                                      class="DuiDialog-root"
                                      style="z-index: 1000; visibility: visible;"
                                    >
                                      <div>
                                        <div
                                          class="DuiDialog-mask"
                                          style="transition: opacity 300ms; opacity: 1; transition-duration: 300ms;"
                                        />
                                        <div
                                          class="DuiDialog-container"
                                        >
                                          <div
                                            class="DuiDialog-paper"
                                            style="width: 50%; margin-top: 15vh; transition: transform 300ms, opacity 300ms; opacity: 1; transform: translate3d(0, 0, 0); transition-duration: 300ms;"
                                          >
                                            <div
                                              class="DuiDialog-close"
                                            >
                                              <button
                                                class="DuiButtonBase-root DuiIconButton-root DuiIconButton-sizeSmall"
                                              >
                                                <span
                                                  class="DuiRipple-root"
                                                />
                                              </button>
                                            </div>
                                            <div
                                              class="DuiDialog-header"
                                            >
                                              <h2>
                                                title
                                              </h2>
                                            </div>
                                            <div
                                              class="DuiDialog-body"
                                            >
                                              Dialog
                                            </div>
                                          </div>
                                        </div>
                                      </div>
                                    </div>
                                    <div
                                      class="DuiDialog-root"
                                      style="z-index: 1000; visibility: visible;"
                                    >
                                      <div>
                                        <div
                                          class="DuiDialog-mask"
                                          style="transition: opacity 300ms; opacity: 1; transition-duration: 300ms;"
                                        />
                                        <div
                                          class="DuiDialog-container"
                                        >
                                          <div
                                            class="DuiDialog-paper"
                                            style="width: 50%; margin-top: 15vh; transition: transform 300ms, opacity 300ms; opacity: 1; transform: translate3d(0, 0, 0); transition-duration: 300ms;"
                                          >
                                            <div
                                              class="DuiDialog-close"
                                            >
                                              <button
                                                class="DuiButtonBase-root DuiIconButton-root DuiIconButton-sizeSmall"
                                              >
                                                <span
                                                  class="DuiRipple-root"
                                                />
                                              </button>
                                            </div>
                                            <div
                                              class="DuiDialog-header"
                                            >
                                              <h2>
                                                title
                                              </h2>
                                            </div>
                                            <div
                                              class="DuiDialog-body"
                                            />
                                          </div>
                                        </div>
                                      </div>
                                    </div>
                                  </body>
                                }
                              >
                                <div
                                  className="DuiDialog-root"
                                  style={
                                    Object {
                                      "visibility": "visible",
                                      "zIndex": 1000,
                                    }
                                  }
                                >
                                  <Transition
                                    appear={true}
                                    enter={true}
                                    exit={true}
                                    in={true}
                                    mountOnEnter={false}
                                    onEnter={[Function]}
                                    onEntered={[Function]}
                                    onEntering={[Function]}
                                    onExit={[Function]}
                                    onExited={[Function]}
                                    onExiting={[Function]}
                                    timeout={300}
                                    unmountOnExit={false}
                                  >
                                    <div>
                                      <div
                                        className="DuiDialog-mask"
                                        style={
                                          Object {
                                            "opacity": 1,
                                            "transition": "opacity 300ms",
                                          }
                                        }
                                      />
                                      <div
                                        className="DuiDialog-container"
                                        onClick={[Function]}
                                      >
                                        <div
                                          className="DuiDialog-paper"
                                          onClick={[Function]}
                                          style={
                                            Object {
                                              "marginTop": "15vh",
                                              "opacity": 1,
                                              "transform": "translate3d(0, 0, 0)",
                                              "transition": "transform 300ms, opacity 300ms",
                                              "width": "50%",
                                            }
                                          }
                                        >
                                          <div
                                            className="DuiDialog-close"
                                            onClick={[Function]}
                                          >
                                            <IconButton
                                              centerRipple={true}
                                              color="default"
                                              size="small"
                                            >
                                              <ButtonBase
                                                centerRipple={true}
                                                className="DuiIconButton-root DuiIconButton-sizeSmall"
                                                component="button"
                                                disableRipple={false}
                                                disabled={false}
                                              >
                                                <button
                                                  className="DuiButtonBase-root DuiIconButton-root DuiIconButton-sizeSmall"
                                                  disabled={false}
                                                >
                                                  <Ripple
                                                    center={true}
                                                  >
                                                    <span
                                                      className="DuiRipple-root"
                                                      onDragLeave={[Function]}
                                                      onMouseDown={[Function]}
                                                      onMouseLeave={[Function]}
                                                      onMouseUp={[Function]}
                                                    />
                                                  </Ripple>
                                                </button>
                                              </ButtonBase>
                                            </IconButton>
                                          </div>
                                          <div
                                            className="DuiDialog-header"
                                          >
                                            <h2>
                                              title
                                            </h2>
                                          </div>
                                          <div
                                            className="DuiDialog-body"
                                          >
                                            Dialog
                                          </div>
                                        </div>
                                      </div>
                                    </div>
                                  </Transition>
                                </div>
                              </Portal>
                            </Portal>
                          </Dialog>
                        </div>
                      </div>
                    </div>
                  </div>
                </Transition>
              </div>
            </Portal>
          </Portal>
        </Dialog>
      </ThemeProvider>
    </JssProvider>
  </ThemeProvider>
</Wrapper>
`;
